tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using StackExchange.Redis;
+using Xunit;
+
+namespace Aspire.Hosting.Redis.Tests;
+
+public class RedisFunctionalTests
+{
+    [Fact]
+    public async Task VerifyRedisResource()
```

Comment:
cc @eerhardt @radical @ReubenBond @mitchdenny thoughts?

Reply:
I think this test adds value over top of the other Redis Hosting tests, since it actually connects a client to it.

It seems like a "middle of the ground" test. Not a unit test, but not a full end-to-end functional test like 

https://github.com/dotnet/aspire/blob/8e311f3892178d441ae2f4f55763b2837fd50d2a/tests/Aspire.EndToEnd.Tests/IntegrationServicesTests.cs#L32-L53

It is definitely simpler than the end-to-end tests.

Reply:
I like these tests, they are a good smoke test. I do think they need to be broken up though. That way if I've made some changes in "redis" land I can run everything related to Redis just by pointing at a particular test assembly.

Reply:
I don't think it's testing less than the current E2E tests, at least when looking from the Redis POV. The E2E is only creating a distinct web api project to use the component, that's the difference with this local functional test.

Do we still need the Redis portion of the E2E test if so? Or should we extract the current portion of the E2E test into a separate project?

For instance, could we not add more volumes testing with this new functional test project already?

Reply:
> I don't think it's testing less than the current E2E tests

The part that the current E2E tests cover that aren't covered here is the "WithReference" portion that connects a separate project with a resource. This is being simulated in the current test with:

```csharp
        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
        {
            ["ConnectionStrings:redis"] = await redis.Resource.GetConnectionStringAsync()
        });
```

> Do we still need the Redis portion of the E2E test if so? Or should we extract the current portion of the E2E test into a separate project?

I think once we have all the AppHost resource tests split into separate test assemblies, and FunctionalTests like this one for each resource, we can remove the bulk of the resources from the E2E test, and only keep maybe ~5 core resources being tested there. That way when a new resource/component gets added we don't need to update the E2E test and have it continually grow, like it is doing today.